I used Python with Biopython online to analyse GenBank files of CRISPR plasmids designed for the project on SRG1 Knockout in Saccharomyces cerevisiae to investigate its role in chromatin regulation and gene expression. This analysis allowed me to:

Extract and visualise key features such as sgRNA guide sequences, PAM sites, promoters, Cas9 coding regions, and terminators.
Validate the correct positioning of essential plasmid components, including sgRNA scaffolds, nuclear localisation signals (NLS), and selectable markers, ensuring Cas9 functionality and sgRNA expression.
Ensure the plasmid sequences align with the experimental requirements for targeting SRG1, with high specificity and minimal off-target effects.
The outputs from Python were cross-referenced with Benchling annotations to confirm the plasmids' integrity and suitability for CRISPR-mediated gene knockout experiments in yeast.

In [1]:
from google.colab import files

# Upload your file
uploaded = files.upload()

Saving crispr-assembly-knock-out-plasmid.gb to crispr-assembly-knock-out-plasmid.gb


In [2]:
file_path = "/content/crispr-assembly-knock-out-plasmid.gb"

In [3]:
# Install Biopython
!pip install biopython

# Import necessary libraries
from Bio import SeqIO

# Load the GenBank file
plasmid_record = SeqIO.read(file_path, "genbank")

# General information
print(f"Plasmid ID: {plasmid_record.id}")
print(f"Sequence Length: {len(plasmid_record.seq)} bp")
print(f"Description: {plasmid_record.description}")

# Extract and display features
print("\nFeatures:")
for feature in plasmid_record.features:
    print(f"Type: {feature.type}, Location: {feature.location}, Strand: {feature.strand}")
    if 'note' in feature.qualifiers:
        print("  Notes:", feature.qualifiers['note'])
    if 'gene' in feature.qualifiers:
        print("  Gene:", feature.qualifiers['gene'])
    if 'product' in feature.qualifiers:
        print("  Product:", feature.qualifiers['product'])

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.2/3.2 MB 39.9 MB/s eta 0:00:00
Plasmid ID: CRISPR_Assembly_Knock-
Sequence Length: 11382 bp
Description: 

Features:
Type: primer, Location: [0:23](+), Strand: 1
  Notes: ['sequence: CACCGAGTCGGTGCTTTTTTatggacaagaagtattctatcgg']
Type: Cas9(N), Location: [0:1719](+), Strand: 1
  Notes: ['codon_start: 1 product: N-terminal portion of Streptococcus pyogenes Cas9 (Zetsche et al., 2015) transl_table: 1 translation: MDKKYSIGLDIGTNSVGWAVITDEYKVPSKKFKVLGNTDRHSIKKNLIGALLFDSGETAEATRLKRTARRRYTRRKNRICYLQEIFSNEMAKVDDSFFHRLEESFLVEEDKKHERHPIFGNIVDEVAYHEKYPTIYHLRKKLVDSTDKADLRLIYLALAHMIKFRGHFLIEGDLNPDNSDVDKLFIQLVQTYNQLFEENPINASGVDAKAILSARLSKSRRLENLIAQLPGEKKNGLFGNLIALSLGLTPNFKSNFDLAEDAKLQLSKDTYDDDLDNLLAQIGDQYADLFLAAKNLSDAILLSDILRVNTEITKAPLSASMIKRYDEHHQDLTLLKALVRQQLPEKYKEIFFDQSKNGYAGYIDGGASQEEFYKFIKPILEKMDGTEELLVKLNREDLLRKQRTFDNGSIPHQIHLGELHAILRRQEDFYPFLKDNREKIEKILTFRIPYYVGPLARGNSRFAWMTRKSEETITPWNFEEVVDKGASAQSFIERMTNFDKNLPNEKVLPKHSLLYEYFTVYNELTKVKYVTEGMRKPAFL

/usr/local/lib/python3.10/dist-packages/Bio/SeqFeature.py:1041: BiopythonParserWarning: Attempting to fix invalid location '11272..4' as it looks like incorrect origin wrapping. Please fix input file, this could have unintended behavior.
  warnings.warn(
/usr/local/lib/python3.10/dist-packages/Bio/SeqFeature.py:231: BiopythonDeprecationWarning: Please use .location.strand rather than .strand
  warnings.warn(


# Feature	Location	Correct Positioning?
Guide sgRNA	[11277:11297]	✅
PAM Site	[11297:11300]	✅
Guide sgRNA Scaffold	[11300:11382]	✅
SNR52 Promoter	[4567:4836]	✅
GAP Promoter	[10539:11206]	✅
Cas9 CDS	[0:1719]	✅
SV40 NLS	[4113:4134]	✅
ADH1 Terminator	[4161:4349]	✅
Yeast 2μ Origin	[7058:8401]	✅
AmpR	[6066:6858]	✅
URA3	[8878:9682]	✅